python/rmm/rmm/_lib/memory_resource.pyx

```diff
@@ -1324,3 +1330,87 @@ def available_device_memory():
     cdef pair[size_t, size_t] res
     res = c_available_device_memory()
     return (res.first, res.second)
+
+
+cdef dict byte_sizes = {
+    "kB": 10 ** 3,
+    "MB": 10 ** 6,
+    "GB": 10 ** 9,
+    "TB": 10 ** 12,
+    "PB": 10 ** 15,
+    "KiB": 2 ** 10,
+    "MiB": 2 ** 20,
+    "GiB": 2 ** 30,
+    "TiB": 2 ** 40,
+    "PiB": 2 ** 50,
+    "B": 1,
+    "": 1,
+}
+
+byte_sizes.update({k.lower(): v for k, v in byte_sizes.items()})
+byte_sizes.update({k[0]: v for k, v in byte_sizes.items() if k and "i" not in k})
+byte_sizes.update({k[:-1]: v for k, v in byte_sizes.items() if k and "i" in k})
+
+cdef int parse_bytes(object s):
+    """ Parse byte string to numbers
+
+    Parameters
+    ----------
+    s : int | str
+        Size of the pool in bytes (shorthand)
+
+    >>> parse_bytes('100')
+    100
+    >>> parse_bytes('100 MB')
+    100000000
+    >>> parse_bytes('100M')
+    100000000
+    >>> parse_bytes('5kB')
+    5000
+    >>> parse_bytes('5.4 kB')
+    5400
+    >>> parse_bytes('1kiB')
+    1024
+    >>> parse_bytes('1e6')
+    1000000
+    >>> parse_bytes('1e6 kB')
+    1000000000
+    >>> parse_bytes('MB')
+    1000000
+    >>> parse_bytes(123)
+    123
+    >>> parse_bytes('5 foos')  # doctest: +SKIP
+    ValueError: Could not interpret 'foos' as a byte unit
+    """
+    cdef double n
+    cdef int multiplier
+    cdef int result
+
+    if isinstance(s, (int, float)):
+        return int(s)
+
+    s = str(s).replace(" ", "")
+    if not s[0].isdigit():
+        s = "1" + s
+
+    cdef int i
+    for i in range(len(s) - 1, -1, -1):
+        if not s[i].isalpha():
+            break
+    index = i + 1
+
+    prefix = s[:index]
+    suffix = s[index:]
```

Comment:
*suggestion*: Although they normally mean you now have two problems, it feels like this is a reasonable place to employ a regular expression.


```
pattern = re.compile(r"^([0-9]+(?:\.[0-9]*)?)[\t ]*((?i:(?:[kmgtp]i?)?b))?$")
match = pattern.match(s)
if match is None:
    raise ValueError(f"Could not parse {s} as a byte specification")
n = float(match.group(1))
suffix = match.group(2) # will be None if bare number
multiplier = {None: 1, "b": 1, "kb": 1000, "mb": 1000_000, ...}[suffix.lower()]
return int(n * multiplier)
```

WDYT?

Reply:
Applied your suggestion